fenixedu-ist-integration/src/main/java/pt/ist/fenixedu/integration/api/FenixAPIv1.java

```diff
@@ -1078,6 +1081,46 @@ public String enrolledByDegreeAndCurricularYear(@PathParam("id") String degreeId
         }).collect(StreamUtils.toJsonArray()).toString();
     }
 
+    @GET
+    @Produces(JSON_UTF8)
+    @Path("users")
+    @OAuthEndpoint(value = PERSONAL_SCOPE, serviceOnly = true)
+    public String getUsers() {
+        Stream<User> users = Bennu.getInstance().getUserSet().stream();
```

Comment:
This seems like a dangerous endpoint. It will generate quite a large output, causing large processing times in both applications.

Might I suggest a web-hook like model, where the target application generates a URL for which this information will be POST'ed? Or returning this in chunks?